.github/scripts/check-folders.js

```diff
@@ -4,63 +4,81 @@ const yaml = require('yaml')
+  ]
+  
+  // Keys to ignore
+  // TODO better description what this actually means
+  const keysToIgnore = [
+    'os', // We want to count folders vs references in the yaml file not when something is run across differnt OSes
+    'node', // We want to count folders vs references in the yaml file not when something is run across differnt node versions
+  ]
+  
+  // Get all relevant folders that _should_ appear in workflows
   const folders = glob
     .sync('**/package.json', {
       ignore: ['**/node_modules/**', '**/custom-engines/**'],
     })
     .map((file) => file.replace('/package.json', ''))
     .filter((file) => {
-      const ignoreFiles = [
-        'package.json', // package.json at root
-        'platforms-serverless/vercel-with-redwood/api', // Redwood uses workspaces but is included
-        'platforms-serverless/vercel-with-redwood/web', // Redwood uses workspaces but is included
-        'platforms-serverless/firebase-functions/functions', // Firebase root doesn't have package.json but is included
-        'generic/basic', // generic/basic doesn't use Github action matrix feature which we parse to find out the differences
-        'packagers/yarn-workspaces/prisma-project', // Yarn workspaces root doesn't have package.json but is included
-        'packagers/yarn3-workspaces-pnp/packages/sub-project-1', // We don't want to include the workspace folders in the matrix
-        'packagers/yarn3-workspaces-pnp/packages/sub-project-2', // We don't want to include the workspace folders in the matrix
-        'platforms/aws-graviton/code', // aws-graviton doesn't have package.json at root but is included
-        'platforms/m1-macstadium/code', // m1-macstadium doesn't have package.json at root but is included
-      ]
       return !ignoreFiles.includes(file)
     })
 
+  // get optional tests workflow
   const optionalTestYamlString = fs.readFileSync(
     path.join(process.cwd(), '.github/workflows/optional-test.yaml'),
     { encoding: 'utf8' },
   )
   const optionalTestYaml = yaml.parse(optionalTestYamlString)
 
+  // get tests workflow
   const testYamlString = fs.readFileSync(
     path.join(process.cwd(), '.github/workflows/test.yaml'),
     { encoding: 'utf8' },
   )
   const testYaml = yaml.parse(testYamlString)
 
+  // get job keys from workflow files
   const jobKeys = Object.keys(optionalTestYaml['jobs'])
     .concat(Object.keys(testYaml['jobs']))
     .filter((key) => {
-      const jobsToIgnore = [
-        'start-time', // Not a test but a job that fills an env var with the job start time
-        'report-to-slack-success', // Not a test but a job that posts to slack
-        'report-to-slack-failure', // Not a test but a job that posts to slack
-        'detect_jobs_to_run', // Not a test but a job that decides which tests should run
-        'cleanup-runs', // Not a test but a job that cancels previous runs
-      ]
       return !jobsToIgnore.includes(key)
     })
+  
+  // TODO ??
```

Comment:
Create an array of job names, from reading the `jobs` keys of the .yml file

Example of an item should look like `platforms-serverless`

Reply:
```suggestion
  // Create an array of job names, from reading the `jobs` keys of the .yml file
```